src/metatensor/models/soap_bpnn/train.py

```diff
@@ -1,40 +1,60 @@
 import logging
 
 import torch
+from omegaconf import OmegaConf
+
+from metatensor.models import ARCHITECTURE_CONFIG_PATH
 
 from ..utils.composition import calculate_composition_weights
 from ..utils.data import collate_fn
+from ..utils.model_io import save_model
+from .model import ARCHITECTURE_NAME
+
+
+DEFAULT_TRAINING_HYPERS = OmegaConf.to_container(
+    OmegaConf.load(ARCHITECTURE_CONFIG_PATH / "soap_bpnn.yaml")
+)["training"]
+
+logger = logging.getLogger(__name__)
 
 
 def loss_function(predicted, target):
     return torch.sum((predicted.block().values - target.block().values) ** 2)
 
 
-logger = logging.getLogger(__name__)
-
+def train(model, train_dataset, hypers=DEFAULT_TRAINING_HYPERS):
+    model_hypers = hypers["model"]
```

Comment:
why do we need the `model_hypers` here?

Reply:
Because the model is also saved during training (this is known as checkpointing, so if anything goes wrong, you can restart the training from a checkpoint or use the checkpoint). In order to save the model during training, we need all of its metadata (including hypers) so that we can initialize a new instance of the model with the same hypers

Reply:
I know what checkpointing is and what we want to achieve but for me it seems we provide the same argument twice. See my other point below.